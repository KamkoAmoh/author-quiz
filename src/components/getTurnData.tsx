import { shuffle, sample} from 'underscore';
import '../index.css';

const getTurnData = (authors) => {
  const allBooks = authors.reduce((p, c) => {
    return p.concat(c.books);
  }, []);

  const fourRandomBooks = shuffle(allBooks).slice(0, 4);
  const answer = sample(fourRandomBooks);

  return {
    books: fourRandomBooks,
    author: authors.find((author) => author.books.some((title) => title === answer))
  }
}

export default getTurnData;
