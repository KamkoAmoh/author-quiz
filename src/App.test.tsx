import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from './components/display/Display.Components';
import authors  from './data/Data.authors';
import Hero from './components/display/display.components/Diplay.Hero';
import renderer from 'react-test-renderer';
import Continue from './components/display/display.components/Display.Continue';
import { Simulate } from 'react-dom/test-utils';
import Book from './components/display/display.components/Display.Book';
import Turn from './components/display/display.components/Display.Turn';
import * as Components from './components/display/Display.Components';
import getTurnData from './components/getTurnData';

test('Get footer text', () => {
  const { getByText } = render(<Footer />);
  const linkElement = getByText(/Wikimedia Commons/ig);
  expect(linkElement).toBeInTheDocument();
});

test('Expect length to be greater than 0', () => {
  expect(authors.length).toBeGreaterThan(0);
});

test('Expect to reder:', () => {
    const rendered = renderer.create(<Hero />).toJSON();
    expect(rendered).toMatchSnapshot();
});

test('Expect Continue to be shown and clickable', () => {
  const c = 'correct';
  const rendered = renderer.create(<Continue show={c} onContinue={Simulate.click} />);
  expect(rendered.toJSON()).toBeTruthy();
});

test('Expect Book to be rendered with items', () => {
  const title = "test";
  const rendered = renderer.create(<Book title={title} onClick={Simulate.click} />);
  expect(rendered.toJSON()).toBeTruthy();
});

test('Expect Turn to have attributes and render', () => {
  const turnData = {
    author: "name",
    books: ["book", "book2"]
  }
  const hightlight = 'correct';
  const rendered = renderer.create(<Turn {...turnData} highlight={hightlight} onAnswerSelected={Simulate.click} />).toJSON();
  expect(rendered).toMatchSnapshot();
});

test("Should have access to all Display Components", () => {
  const title = "Book Title";
  const turnData = {
    author: "name",
    books: ["book", "book1"]
  }
  const hgh = 'correct';
  const rendered = renderer.create(
    <div>
      <Components.Book title={title} onClick={Simulate.click} />
      <Components.Continue show={hgh} onContinue={Simulate.click} />
      <Components.Turn {...turnData} highlight={hgh} onAnswerSelected={Simulate.click} />
      <Components.Footer />
      <Components.Hero />
    </div>
  ).toJSON();
  expect(rendered).toMatchSnapshot();
});

test("Should receive turn Data", () => {
  const authors = [
    {
      name: "name",
      imageUrl: "imageUrl",
      books: ["book", "book2"]
    },
    {
      name: "name1",
      imageUrl: "imageUrl1",
      books: ["book3", "book1"]
    }
  ];
  expect(getTurnData(authors)).toBeTruthy();
});


