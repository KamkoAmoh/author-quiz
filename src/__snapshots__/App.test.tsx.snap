// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect Turn to have attributes and render 1`] = `
<div
  className="turn__content"
  style={
    Object {
      "backgroundColor": "lightGreen",
    }
  }
>
  <div
    className="turn_image"
  >
    <img
      alt="Author"
      className="authorimage"
    />
  </div>
  <div
    className="turn__answer__row"
  >
    <div
      className="answer"
      onClick={[Function]}
    >
      <h4>
        book
      </h4>
    </div>
    <div
      className="answer"
      onClick={[Function]}
    >
      <h4>
        book2
      </h4>
    </div>
  </div>
</div>
`;

exports[`Expect to reder: 1`] = `
<div
  className="hero"
>
  <div
    className="hero__content"
  >
    <h1>
      Auhtor Quiz
    </h1>
    <p>
      Select the book written by:
    </p>
  </div>
</div>
`;

exports[`Should have access to all Display Components 1`] = `
<div>
  <div
    className="answer"
    onClick={[Function]}
  >
    <h4>
      Book Title
    </h4>
  </div>
  <div
    className="continue"
  >
    <div
      className="col-11"
    >
      <button
        className="btn btn-primary btn-lg float-right"
        onClick={[Function]}
      >
        Continue
      </button>
    </div>
  </div>
  <div
    className="turn__content"
    style={
      Object {
        "backgroundColor": "lightGreen",
      }
    }
  >
    <div
      className="turn_image"
    >
      <img
        alt="Author"
        className="authorimage"
      />
    </div>
    <div
      className="turn__answer__row"
    >
      <div
        className="answer"
        onClick={[Function]}
      >
        <h4>
          book
        </h4>
      </div>
      <div
        className="answer"
        onClick={[Function]}
      >
        <h4>
          book1
        </h4>
      </div>
    </div>
  </div>
  <div
    className="footer"
    id="footer"
  >
    <div
      className="footer__frame"
    >
      <p
        className="footer__content"
      >
        All images are from 
        <a
          href="https://commons.wikimedia.org/wiki/Main_Page"
        >
          Wikimedia Commons
        </a>
         and are in public domain.
      </p>
    </div>
  </div>
  <div
    className="hero"
  >
    <div
      className="hero__content"
    >
      <h1>
        Auhtor Quiz
      </h1>
      <p>
        Select the book written by:
      </p>
    </div>
  </div>
</div>
`;
